src/SELF_MappedScalar_2D_t.f90

```diff
@@ -118,174 +118,172 @@ subroutine SetInteriorFromEquation_MappedScalar2D_t(this,geometry,time)
-  !                              (this%interp%N+1), &
-  !                              decomp%mpiPrec, &
-  !                              r2,globalSideId, &
-  !                              decomp%mpiComm, &
-  !                              decomp%requests(msgCount),iError)
-
-  !               msgCount = msgCount+1
-  !               call MPI_ISEND(this%boundary(:,s1,e1,ivar), &
-  !                              (this%interp%N+1), &
-  !                              decomp%mpiPrec, &
-  !                              r2,globalSideId, &
-  !                              decomp%mpiComm, &
-  !                              decomp%requests(msgCount),iError)
-  !             endif
-  !           endif
-
-  !         enddo
-  !       enddo
-  !     enddo
-
-  !     decomp%msgCount = msgCount
-  !   endif
-
-  ! endsubroutine MPIExchangeAsync_MappedScalar2D_t
-
-  ! subroutine ApplyFlip_MappedScalar2D_t(this,decomp,mesh)
-  !   ! Apply side flips to sides where MPI exchanges took place.
-  !   implicit none
-  !   class(MappedScalar2D_t),intent(inout) :: this
-  !   type(MPILayer),intent(inout) :: decomp
-  !   type(Mesh2D),intent(in) :: mesh
-  !   ! Local
-  !   integer :: e1,s1,e2,s2
-  !   integer :: i,i2
-  !   integer :: r2,flip,ivar
-  !   integer :: globalSideId
-  !   integer :: bcid
-  !   real(prec) :: extBuff(1:this%interp%N+1)
-
-  !   if(decomp%mpiEnabled) then
-  !     do ivar = 1,this%nvar
-  !       do e1 = 1,this%nElem
-  !         do s1 = 1,4
-
-  !           e2 = mesh%sideInfo(3,s1,e1) ! Neighbor Element
-  !           s2 = mesh%sideInfo(4,s1,e1)/10
-  !           bcid = mesh%sideInfo(5,s1,e1)
-  !           if(e2 /= 0) then ! Interior Element
-  !             r2 = decomp%elemToRank(e2) ! Neighbor Rank
-
-  !             if(r2 /= decomp%rankId) then
-
-  !               flip = mesh%sideInfo(4,s1,e1)-s2*10
-  !               globalSideId = mesh%sideInfo(2,s1,e1)
-
-  !               ! Need to update extBoundary with flip applied
-  !               if(flip == 1) then
-
-  !                 do i = 1,this%interp%N+1
-  !                   i2 = this%interp%N+2-i
-  !                   extBuff(i) = this%extBoundary(i2,s1,e1,ivar)
-  !                 enddo
-  !                 do i = 1,this%interp%N+1
-  !                   this%extBoundary(i,s1,e1,ivar) = extBuff(i)
-  !                 enddo
-
-  !               endif
-  !             endif
-
-  !           endif
-
-  !         enddo
-  !       enddo
-  !     enddo
-
-  !   endif
-
-  ! endsubroutine ApplyFlip_MappedScalar2D_t
-
-  subroutine SideExchange_MappedScalar2D_t(this,mesh,decomp)
+  subroutine MPIExchangeAsync_MappedScalar2D_t(this,mesh,resetCount)
+    implicit none
+    class(MappedScalar2D_t),intent(inout) :: this
+    type(Mesh2D),intent(inout) :: mesh
+    logical,intent(in) :: resetCount
+    ! Local
+    integer :: e1,s1,e2,s2,ivar
+    integer :: globalSideId,r2,tag
+    integer :: iError
+    integer :: msgCount
+
+    if(resetCount) then
```

Comment:
Remove the conditional on resetcount, and remove this input from the API. For now, we'll assume each exchange case is isolated and we will always reset the count.

Reply:
Resolved in [f24f99d](https://github.com/FluidNumerics/SELF/pull/53/commits/f24f99db9e33d33c61cba24a30e9e8176c9a85ea)